### PR TITLE
Add dependent ops to BaseModel._create_train_ops

### LINKED ISSUE #66: `cxflow resume` fails if `model.n_gpus` is less than the original value
This may be related to batch normalization routines.

### PR DESCRIPTION
**Pull request by blazekadam**
- this allows to use `BatchNormalization`
- supports multi-GPU but requires `allow_soft_placement=true` in such case

closes #66

### CODE REVIEW

File: cxflow_tensorflow/model.py [hunk 1]
```diff
@@ -198,19 +198,21 @@ def __init__(self,  # pylint: disable=too-many-arguments
...
         self._session = self._create_session(session_config)
+        dependencies = []
         with self._graph.as_default():
             if restore_from is None:
                 with tf.variable_scope(tf.get_variable_scope()) as scope:
                     self._is_training = tf.placeholder(tf.bool, [], BaseModel.TRAINING_FLAG_NAME)
                     for tower in self._towers:
                         with tower:
                             self._create_model(**kwargs)
+                        dependencies.append([]+self._graph.get_collection(tf.GraphKeys.UPDATE_OPS))
```
[the remark this answers is not in the document]
Comment (blazekadam) [reply cxflow_tensorflow/model.py]: A copy of that `list` is needed. Does `list()` perform a copy?

[the remark this answers is not in the document]
Comment (FloopCZ) [reply cxflow_tensorflow/model.py]: Yes, it does. From [the docs](https://docs.python.org/3/library/stdtypes.html#list):
>  If iterable is already a list, a copy is made and returned

[the remark this answers is not in the document]
Comment (blazekadam) [reply cxflow_tensorflow/model.py]: Okay I ll make it less funky once we merge #69 and #68 ...

**Comment by FloopCZ** [comment]
@blazekadam Great, thanks. Feel free to merge whenever you are ready.
